pkg/apis/batch/validation/validation.go

```diff
@@ -182,6 +192,15 @@ func validateJobSpec(spec *batch.JobSpec, fldPath *field.Path, opts apivalidatio
 	if spec.TTLSecondsAfterFinished != nil {
 		allErrs = append(allErrs, apivalidation.ValidateNonnegativeField(int64(*spec.TTLSecondsAfterFinished), fldPath.Child("ttlSecondsAfterFinished"))...)
 	}
+	if spec.BackoffLimitPerIndex != nil {
```

Comment:
Not sure I understand. The code you mark validates the `BackoffLimitPerIndex` when it is set. In that case it needs to be non-negative etc. 

I have thought about downgrade scenario, and I think it works ok. Let's suppose the feature is Beta and a user has set `BackoffLimitPerIndex=1`. Now, the system is downgraded to Alpha, and the feature gate is disabled. In that case the field is preserved, see here: https://github.com/kubernetes/kubernetes/blob/781fee383002975cb2351f5d26816062c3e9586e/pkg/registry/batch/job/strategy.go#L122

Then, it is validated in the line pointed out in this comment. The validation will still pass.

Reply:
So in the batch validation we have this struct: https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/batch/validation/validation.go#L587

I was thinking it would be nice to consolidate the logic I mentioned there.  

ie we should only do validation if feature toggle is on but I think there is a requirement.

@alculquicondor pointed this pattern to me when doing the prefixed label work.  Not sure if we should keep the same pattern here.

Reply:
I know about this structure, I think it is used when the validation for updates depends on the old object. For example, it is used for `AllowMutableSchedulingDirectives` and `RequirePrefixedLabels`, which depend on the old object: https://github.com/kubernetes/kubernetes/blob/master/pkg/registry/batch/job/strategy.go#L153-L163

I guess I still could use this pattern, but it is not obvious to me if this is just consistency or I would solve some issue.

Reply:
Thanks for clearing that up and pointing that out.  Didn't realize that was for validation of updates.  Makes sense now.